### PR TITLE
convert magic zoom to web api

### PR DESCRIPTION
**Pull request by steveoh**
so far without tests.

### CODE REVIEW

File: widgets/locate/MagicZoom.js [hunk 1]
```diff
@@ -173,12 +174,9 @@ define([
             this._setUpQueryTask();
             this._wireEvents();
             this._setUpGraphicsLayer();
-        },
-        showSpinner: function() {
-            // summary:
-            //      sets up and shows the spinner
-            console.log('agrc.widgets.locate.MagicZoom::showSpinner', arguments);
-
+            this.webApi = new WebApi({
```
Comment (stdavis) [inline widgets/locate/MagicZoom.js]: Add `webApi` to list of constructor parameters on line 166-ish.

Comment (steveoh) [inline widgets/locate/MagicZoom.js]: addressed: https://github.com/agrc/agrc.widgets/commit/bb0d33c7f467a5115f57c4212ccbaf18781ca5ee

**Comment by stdavis** [comment]
Test page needs to be fixed.

**Comment by stdavis** [comment]
Once the tests are passing (sauce was having server issues earlier) this looks great to me.

**Comment by stdavis** [comment]
@steveoh Did we have a solution for searching layers that are not in SGID?

**Comment by steveoh** [comment]
nope

**Comment by stdavis** [comment]
Super helpful. Thanks! 

I guess I'll bring the old one back to life? What way do you want me to do it so that it works with your grand scheme for widgets? I need the ability to search a layer within a map service that points to data not in SGID.

**Comment by steveoh** [comment]
i think you could branch the fetch logic if the widget is instantiated with a url and id combo.

**Comment by stdavis** [comment]
OK. I'll do that. Did you want to split out this widget into it's own repo as well?

**Comment by steveoh** [comment]
Yes, this widget should make the cut. I would suggest creating a new name and incorporating any upgrades/breaking changes/public api cleanup. It might be nice to give this widget the idea of providers. Where the web api or a map service is a provider/source/store. That way you could build out any type of provider you need and pass that to the zoomie findie drop down suggestion widget.

**Pull request merged**